src/algorithms/superviewport.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AbstractViewportAlgorithm,
+  AlgorithmInput,
+  AlgorithmOutput,
+} from "./core";
+import SuperCluster, { ClusterFeature } from "supercluster";
+import { MarkerUtils, Marker } from "../marker-utils";
+import { Cluster } from "../cluster";
+import { getPaddedViewport } from "./utils";
+import equal from "fast-deep-equal";
+
+export type SuperClusterViewportOptions = SuperCluster.Options<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  { [name: string]: any },
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  { [name: string]: any }
+> & { viewportPadding?: number };
+
+export interface SuperClusterViewportState {
+  zoom: number;
+  view: [number, number, number, number];
```

Comment:
suggest adding a comment on what the numbers are

Reply:
Done